interfaces/builtin/dbus.go

```diff
@@ -428,8 +437,49 @@ func (iface *dbusInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
 }
 
 func (iface *dbusInterface) BeforePrepareSlot(slot *snap.SlotInfo) error {
-	_, _, err := iface.getAttribs(slot)
-	return err
+	bus, name, err := iface.getAttribs(slot)
+	if err != nil {
+		return err
+	}
+
+	var apps []*snap.AppInfo
+	for _, app := range slot.Apps {
+		for _, slotName := range app.ActivatesOn {
+			if slotName == slot.Name {
+				apps = append(apps, app)
+				break
+			}
+		}
+	}
+	if len(apps) > 1 {
+		return fmt.Errorf("cannot add activatable dbus service slot to multiple apps")
+	}
+	if len(apps) == 1 {
+		app := apps[0]
+		if !app.IsService() {
+			return fmt.Errorf("only daemons can be activatable D-Bus services")
```

Comment:
This made me realize that the proper spelling is indeed `D-Bus` and not DBus. We could later do a pass and make the tree consistent.